crates/ide-assists/src/handlers/toggle_async_sugar.rs

```diff
@@ -0,0 +1,460 @@
+use ide_db::{
+    assists::{AssistId, AssistKind},
+    famous_defs::FamousDefs,
+};
+use syntax::{
+    ast::{self, HasVisibility},
+    AstNode, NodeOrToken, SyntaxKind, SyntaxNode, SyntaxToken, TextRange,
+};
+
+use crate::{AssistContext, Assists};
+
+// Assist: toggle_async_sugar
```

Comment:
I'd prefer if we split the assist into two distinct labeled ones (still keep them in the same module here as they are short enough). `desugar_async_function` and `resugar_async_function` or similar, and then break apart the `toggle_async_sugar` function into two entry points that are called in the `lib.rs` file

Reply:
Changed, now it's `desugar_async_into_impl_future` and `sugar_impl_future_into_async`, other names were a bit ambiguous to me